src/parser/smt2/Smt2.g

```diff
@@ -713,11 +713,20 @@ setOptionInternal[std::unique_ptr<cvc5::parser::Command>* cmd]
   cvc5::Term sexpr;
 }
   : keyword[name] symbolicExpr[sexpr]
-    { cmd->reset(new SetOptionCommand(name.c_str() + 1, sexprToString(sexpr)));
+    { 
+      std::string key = name.c_str() + 1;
+      std::string ss = sexprToString(sexpr);
+      // special case: for channel settings, we are expected to parse e.g.
+      // `"stdin"` which should be treated as `stdin`
+      if (key=="diagnostic-output-channel" || key=="regular-output-channel" || key=="in")
```

Comment:
Instead of matching on specific names, I think that it would be cleaner to use [`getOptionInfo`](https://github.com/cvc5/cvc5/blob/main/src/api/cpp/cvc5.h#L4227) and check whether the expected argument is a string (or maybe a stream? The current `OptionInfo` type may have to be expanded). This would ensure that we treat all string options the same and also that we treat the options and their aliases the same (the current solution does something different for `:out` and `:regular-output-channel` even though they map to the same option).